### PR TITLE
[PF-2256] Add tag and publish workflow

### PR DESCRIPTION
**Pull request by iliadsh**
So that the client can be used in other projects.

Needs:
BROADBOT_TOKEN
ARTIFACTORY_USERNAME
ARTIFACTORY_PASSWORD

### CODE REVIEW

File: .github/workflows/tag-publish.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+name: Bump, Tag, Publish
+
+on:
+  pull_request:
```
Comment (yuhuyoyo) [inline .github/workflows/tag-publish.yml]: are you removing this part after testing? 

i don't think we need to bump version on PR

File: .github/workflows/tag-publish.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    
+    - name: "Publish to Artifactory"
+      run: ./gradlew --build-cache :client:artifactoryPublish --scan
+      env:
+        ARTIFACTORY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
+        ARTIFACTORY_PASSWORD: ${{ secrets.ARTIFACTORY_PASSWORD }}
+        ARTIFACTORY_REPO_KEY: "libs-release-local"
```
Comment (yuhuyoyo) [inline .github/workflows/tag-publish.yml:80]: do you need to push a docker image too? for reference, i'm looking at the template repo for this GHA https://cs.github.com/DataBiosphere/terra-java-project-template/blob/e819037eb3c41e277f182634679d7399935ea801/.github/workflows/publish.yml#L32-L59.

File: client/build.gradle [hunk 3]
```diff
@@ -8,11 +8,7 @@ plugins {
     id 'org.hidetake.swagger.generator'
 }
 
-dependencyManagement {
-    imports {
-        mavenBom(SpringBootPlugin.BOM_COORDINATES)
-    }
-}
+repositories { mavenCentral() }
```
Comment (yuhuyoyo) [inline client/build.gradle]: not related?

File: .github/workflows/tag-publish.yml (hunk 1 above)
Comment (iliadsh) [reply .github/workflows/tag-publish.yml]: yes removing, just for testing while here

File: .github/workflows/tag-publish.yml (hunk 2 above)
Comment (iliadsh) [reply .github/workflows/tag-publish.yml:80]: AFAIK the Broad hasn't asked for a deployment of this so I didn't think that was needed

**Review by yuhuyoyo — APPROVED** [review]

File: client/build.gradle (hunk 3 above)
Comment (iliadsh) [reply client/build.gradle]: trying to make the pom generation happy :(

**Review by cbookg — APPROVED** [review]

**Pull request merged**